depends/packages/qt.mk

```diff
@@ -24,13 +24,14 @@ define $(package)_set_vars
 $(package)_config_opts_release = -release
 $(package)_config_opts_debug = -debug
 $(package)_config_opts += -bindir $(build_prefix)/bin
-$(package)_config_opts += -c++std c++11
+$(package)_config_opts += -c++std c++17
```

Comment:
Qt configure reports an error on Xenial build:
```
ERROR: Feature 'c++1z' was enabled, but the pre-condition 'features.c++17' failed.
```

Reply:
Ubuntu 16.04 is not listed as [supported by Qt 5.15.x](https://doc.qt.io/qt-5/supported-platforms.html). Regardless, if Qts build system cant detect that your compiler supports C++17 it's not going to build in C++17 mode. Looking at the GCC on Xenial, it's [5.3.1](https://packages.ubuntu.com/xenial/gcc), [which only has support for some C++17 features](https://gcc.gnu.org/projects/cxx-status.html#cxx17).

If you post the actual precondition that failed (it'll be earlier in the log) we could take a look at it.

Reply:
> If you post the actual precondition that failed (it'll be earlier in the log) we could take a look at it.

https://travis-ci.org/github/hebasto/bitcoin/jobs/720916965

Reply:
```
Checking for C++17 support... 
+ cd /home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/config.tests/cxx17 && /home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/bin/qmake "CONFIG -= qt debug_and_release app_bundle lib_bundle" "CONFIG += static warn_off console single_arch" -early "CONFIG += cross_compile" /home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/config.tests/cxx17
+ cd /home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/config.tests/cxx17 && MAKEFLAGS= /usr/bin/make
> make[1]: Entering directory '/home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/config.tests/cxx17'
> g++ -c -m64 -pipe -pipe -O2 -I/home/travis/build/hebasto/bitcoin/depends/x86_64-pc-linux-gnu/include -O2 -std=gnu++1z -w -fPIC  -I. -I/home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/mkspecs/linux-g++-64 -o main.o main.cpp
> main.cpp:8:19: fatal error: variant: No such file or directory
> compilation terminated.
> Makefile:171: recipe for target 'main.o' failed
> make[1]: *** [main.o] Error 1
> make[1]: Leaving directory '/home/travis/build/hebasto/bitcoin/depends/work/build/x86_64-pc-linux-gnu/qt/5.15.0-e4e26f06c7b/qtbase/config.tests/cxx17'
test config.qtbase.tests.cxx17 FAILE
```

Reply:
> fatal error: variant: No such file or directory

The problem is that the compiler doesn't support C++17. GCC didn't introduce support for `std::variant` (in libstdc++) until the [7.1 release](https://gcc.gnu.org/gcc-7/changes.html). Xenial ships with 5.3.1. So when Qts build system tests for C++17 support by checking it can compile something using `std::variant`, it's failing as expected.